### PR TITLE
authentication models and signup views

### PR DESCRIPTION
**Pull request by yomnaafifi**
created custom user model, forms and signup views for both types of users

### CODE REVIEW

File: waddi/settings.py [hunk 1]
```diff
@@ -102,13 +103,20 @@
...
 DATABASES = {
+
     'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': BASE_DIR / 'db.sqlite3',
+        'ENGINE': 'django.db.backends.postgresql',
+        'NAME': 'waddi',
+        'USER': 'waddi',
+        'PASSWORD': 'password',
+        'HOST': '127.0.0.1',  # Docker container name
```
Comment (IxBahy) [inline waddi/settings.py]: this is not the name

File: authentication/forms.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        fields = ('email','first_name','last_name','birthdate','image','phone_no')
+        
+
+    @transaction.atomic
+    def save(self):
+        user = super().save(commit=False)
+        user.is_customer = True
+        user.save()
+        customer = Customer.objects.create(user=user)
+        customer.preferred_method = self.Preferred_method
```
Comment (IxBahy) [inline authentication/forms.py]: test the value coming here more than once cuz I have doubts it will work

File: authentication/managers.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        if not re.match(email_regex, email):
+            raise ValueError("Invalid Email format")
+        
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, password, **extra_fields):
```
Comment (IxBahy) [inline authentication/managers.py]: if u will pass with comments just delete it

File: authentication/models.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        pattern = r'^01[0-2]\d{1}-\d{3}-\d{4}$'  
+        if not re.match(pattern, value):
+            raise ValidationError(_("Phone number must be in the format 01X-XXX-XXXX"))
+
+    email = models.EmailField(unique = True)
+    first_name = models.CharField(max_length=100)
+    last_name = models.CharField(max_length=100)
+    birthdate = models.DateField()
+    image = models.ImageField(upload_to=None) # further edits needed 
+    phone_no = models.CharField(max_length=100, validators=[validate_phone_number])
```
Comment (IxBahy) [inline authentication/models.py:48]: write the full name

File: customer/serializer.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from rest_framework.serializers import ModelSerializer
+from customer.models import Customer
+
+class CreateCustomerSerializer(ModelSerializer):
+    class Meta:
+        model = Customer
+        fields = ['email','first_name','last_name','birthdate','image','phone_no']
```
Comment (IxBahy) [inline customer/serializer.py]: this will not work

Comment (IxBahy) [reply customer/serializer.py]: this data will come from the CustomUser model 
so fetch the data from the FK

File: customer/views.py [hunk 6]
```diff
@@ -1,3 +1,17 @@
...
+from rest_framework.response import Response
+from customer.models import Customer
+from customer.serializer import CreateCustomerSerializer  
+from authentication.forms import CustomerSignupForm
 
-# Create your views here.
+class CustomerSignupView(generics.CreateAPIView):
+    serializer_class = CreateCustomerSerializer 
+    def post(self, request, *args, **kwargs):
+        form = CustomerSignupForm(request.POST)
```
Comment (IxBahy) [inline customer/views.py]: use a serializer here instead of the form

File: driver/views.py [hunk 7]
```diff
@@ -1,3 +1,15 @@
...
-
-# Create your views here.
+from rest_framework import generics, status
+from driver.serializer import CreateDriverSerializer
+from authentication.forms import DriverSignupForm
+from rest_framework.response import Response
+class DriverSignupView(generics.CreateAPIView):
+    serializer_class = CreateDriverSerializer
+    def post(self, request, *args, **kwargs):
+        form = DriverSignupForm
```
Comment (IxBahy) [inline driver/views.py]: same here dump the forms

**Review by IxBahy — CHANGES_REQUESTED** [review]

**Comment by IxBahy** [comment]
for all the forms we are using dump them and refactor to serializers that will take the same data 
we will handle the creation of user+customer/driver in the api view

File: waddi/settings.py (hunk 1 above)
Comment (yomnaafifi) [reply waddi/settings.py]: i know its an old comment

**Pull request merged**